src/v/cluster/producer_state.h

```diff
@@ -204,41 +221,74 @@ class producer_state {
         return model::timestamp(_last_updated_ts.time_since_epoch() / 1ms);
     }
 
-    std::optional<kafka::offset> current_txn_start_offset() const {
-        return _current_txn_start_offset;
-    }
-
     model::producer_identity id() const { return _id; }
 
-    void update_current_txn_start_offset(std::optional<kafka::offset> offset) {
```

Comment:
Its only "describe_producers" API that is broken, I did it so that all changes to the producer state appear in one commit (for reviewers convenience)/